--- binutils-2.29.1/bfd/cpu-m68k.c.orig	2017-11-06 07:12:40.259332482 +0100
+++ binutils-2.29.1/bfd/cpu-m68k.c	2017-11-06 07:12:43.623332409 +0100
@@ -30,7 +30,7 @@
 		     const bfd_arch_info_type *b);
 
 #define N(name, print,d,next)  \
-{  32, 32, 8, bfd_arch_m68k, name, "m68k",print,2,d,bfd_m68k_compatible, \
+{  32, 32, 8, bfd_arch_m68k, name, "m68k",print,1,d,bfd_m68k_compatible, \
    bfd_default_scan, bfd_arch_default_fill, next, 0 }
 
 static const bfd_arch_info_type arch_info_struct[] =
