diff -rupN a/build-config.sub b/build-config.sub
--- a/config.sub	2017-04-30 00:34:32.000000000 +0200
+++ b/config.sub	2017-10-10 21:22:40.599373437 +0200
@@ -235,6 +235,10 @@ case $os in
 	-psos*)
 		os=-psos
 		;;
+	-mintelf | -mintelf[0-9]*)
+		basic_machine=m68k-atari
+		os=-mintelf
+		;;
 	-mint | -mint[0-9]*)
 		basic_machine=m68k-atari
 		os=-mint
@@ -836,6 +840,10 @@ case $basic_machine in
 	miniframe)
 		basic_machine=m68000-convergent
 		;;
+	*mintelf | -mintelf[0-9]*)
+		basic_machine=m68k-atari
+		os=-mintelf
+		;;
 	*mint | -mint[0-9]* | *MiNT | *MiNT[0-9]*)
 		basic_machine=m68k-atari
 		os=-mint
@@ -1535,6 +1543,9 @@ case $os in
 	-xenix)
 		os=-xenix
 		;;
+	-*mintelf | -mintelf[0-9]*)
+		os=-mintelf
+		;;
 	-*mint | -mint[0-9]* | -*MiNT | -MiNT[0-9]*)
 		os=-mint
 		;;
@@ -1814,6 +1825,9 @@ case $basic_machine in
 			-mpw* | -macos*)
 				vendor=apple
 				;;
+			-*mintelf | -mintelf[0-9]*)
+				vendor=atari
+				;;
 			-*mint | -mint[0-9]* | -*MiNT | -MiNT[0-9]*)
 				vendor=atari
 				;;
